docs/xml-sitemaps.md

```diff
@@ -8,160 +8,51 @@ The sitemap index can hold a maximum of 50000 sitemaps, and a single sitemap can
-		$this->name = 'videos';
-		$this->object_type = 'post';
-	}
+add_action( 'init', function () {
+	global $wpseo_sitemaps;
 
-	/**
-	 * Return the list of URLs for the current page.
-	 */
-	public function get_url_list( $page ) {
-		$videos = new WP_Query( [
-			'post_type' => 'video',
-			// phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_numberposts
-			'posts_per_page' => 2000,
-			'fields' => 'ids',
-			'paged' => $page,
-		] );
-
-		$urls = [];
-
-		foreach ( $videos->posts as $video_id ) {
-			$urls[] = get_post_meta( $video_id, 'video_url', true );
-		}
-
-		return $urls;
+	if ( isset( $wpseo_sitemaps ) && ! empty( $wpseo_sitemaps ) ) {
+		$wpseo_sitemaps->register_sitemap( 'video', 'create_video_sitemap' );
 	}
-
-	/**
-	 * Return the maximum number of pages to output for this sitemap.
-	 */
-	public function get_max_num_pages() {
-		$videos = new WP_Query( [
-			'post_type' => 'video',
-			// phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_numberposts
-			'posts_per_page' => 2000,
-			'fields' => 'ids',
-		] );
-
-		return $videos->max_num_pages;
-	}
-
-}
-
-add_filter( 'init', function () : void {
-	$provider = new Video_Sitemap_Provider();
-	wp_register_sitemap_provider( 'video-sitemaps', $provider );
 } );
 ```
 
-The provider will be responsible for getting all sitemaps and sitemap entries, as well as determining pagination.
-
-## Removing Specific Sitemaps
-There are three existing sitemaps providers for standard object types - `posts`, `taxonomies`, and `users`. If you want to remove one of them such as the "users" provider, you can configure the module like so:
-
-```json
-{
-	"extra": {
-		"altis": {
-			"modules": {
-				"seo": {
-					"xml-sitemaps": {
-						"users": false
-					}
-				}
-			}
-		}
-	}
-}
-```
-
-Alternatively, if you need to use more complex logic or work with custom sitemap providers you can use the `wp_sitemaps_add_provider` filter to do so. For example:
+To remove a post type from your sitemaps, use the `wp_seo_sitemap_exclude_post_type` filter. The example below removes a post type registered as `recipe`.
 
 ```php
-add_filter( 'wp_sitemaps_add_provider', function ( $provider, string $name ) {
-	// Only switch off the user sitemaps for subsites on the network.
-	if ( $name === 'users' && ! is_main_site() ) {
-		return false;
-	}
-
-	return $provider;
+add_filter( 'wpseo_sitemap_exclude_post_type', function ( $excluded, $post_type ) {
+	return $post_type === 'recipe';
 }, 10, 2 );
 ```
 
-If you want to disable sitemap generation for a specific post type or taxonomy, use the `wp_sitemaps_post_types` or `wp_sitemaps_taxonomies` filter, respectively.
-
-To disable sitemaps for the `page` post type you would do the following:
-
-```php
-add_filter( 'wp_sitemaps_post_types', function ( array $post_types ) : array {
-	unset( $post_types['page'] );
-	return $post_types;
-} );
-```
+You can similarly exclude specific posts, taxonomies or authors, using built in filters in Yoast SEO.
```

Comment:
The docs for these aren't huge so I still think we should mirror most if not all of it.